doc/widgets/kmeans.rst

```diff
@@ -1,86 +1,105 @@
+:doc:`Paint Data<../orange3/doc/visual-programming/source/widgets/data/paintdata>`. Interactive k-Means widget also produces a data table
+with results of clustering and a table with centroids positions. These data can be inspected with the
+:doc:`Data Table<../orange3/doc/visual-programming/source/widgets/data/datatable>` widget.
+
+.. figure:: images/interactive-kmeans-input.png
+
+Let us demonstrate the working of the widget on *Iris* data set.
 
-.. figure:: images/orangekmeans1.png
+We provide the data using :doc:`File<../orange3/doc/visual-programming/source/widgets/data/file>`. 
+Then we open **Interactive k-Means**. Say, we will demonstrate k-Means on *petal length* and *petal width* attributes, 
+so we set them as *X* and *Y* parameters. We also decided to perform clustering for 3 clusters. This is set as the 
+*Number of centroids*.
 
-Let us demonstrate the working of the widget on *Iris Dataset*.
+.. figure:: images/interactive-kmeans-step1.png
 
-We provide data from data set using **File widget**. Then we open **Interactive k-Means** widget. We decided that we
-will demonstrate k-Menas on *petal length* and *petal width* attributes, so we set them as *X* and *Y* parameters.
-We also decided to perform a clustering in 3 clusters. This is set as *Number of centroids*. After that we get this
-figure.
+If we are not satisfied with positions of centroids we can change them with a click on the **Randomize** button. Then we
+perform the first recomputing of centroids with a click on the **Recompute centroids**. We get the following image.
 
-.. figure:: images/kmeansstep1.png
+.. figure:: images/interactive-kmeans-step2.png
 
-If we are not satisfied with positions of centroids we can change it with click on **Randomize** button. Then we
-perform first recomputing of centroids with click on **Recompute centroids** button. We get following image.
+The next step is to reassign membership of all points to the closest centroid. This is performed with a click on
+the **Reasign membership** button.
 
-.. figure:: images/kmeansstep2.png
+.. figure:: images/interactive-kmeans-step3.png
 
-The next step is to reassign membership of point to the closest centroid. It is performed with click on
-**Reasign membership** button.
+Then we repeat these two steps until the algorithm converges. This is the final result.
 
-.. figure:: images/kmeansstep3.png
+.. figure:: images/interactive-kmeans-step4.png
 
-Then we repeat that two steps until algorithm converges. It is final result.
+Perhaps we are not satisfied with the result because we noticed that maybe classification into 4 clusters would be better.
+So we decided to add a new centroid. We can do this by increasing the number of centroids in the control menu or with a click on
+the position in the graph where we want to place the centroid. We decided to add it with a click. The new centroid is the red one.
```

Comment:
This is my mistake from before. The last sentence would be: 

    The new centroid is the orange one.